### PR TITLE
Support cephfs as underfs of alluxio

### LINKED ISSUE #13159: support Cephfs as UFS of Alluxio
**Is your feature request related to a problem? Please describe.**

Now Alluxio support ceph as an UFS through S3 API which supplied by radosgateway, but we want to support cephfs as an ufs of alluxio

**Describe the solution you'd like**
Create an cephfs ufs module which extended from hdfs ufs module

**Urgency**
High

**Additional context**
With this feature, we can bring alluxio to the kubernetes production env

### PR DESCRIPTION
**Pull request by maobaolong**
Fix #13159

### CODE REVIEW

File: shaded/pom.xml [hunk 1]
```diff
@@ -27,6 +27,7 @@
     <module>hadoop</module>
     <module>ozone</module>
     <module>cosn</module>
+    <module>hcfs-cephfs</module>
```
Comment (apc999) [inline shaded/pom.xml:30]: can you sort these alphebetically.

File: underfs/pom.xml [hunk 2]
```diff
@@ -37,6 +37,7 @@
     <module>web</module>
     <module>ozone</module>
     <module>cosn</module>
+    <module>hcfs-cephfs</module>
```
Comment (apc999) [inline underfs/pom.xml:40]: same here, sort modules alphabetically.

File: dev/scripts/src/alluxio.org/build-distribution/cmd/generate-tarball.go [hunk 3]
```diff
@@ -216,6 +216,7 @@ func addAdditionalFiles(srcPath, dstPath string, hadoopVersion version, version
 		"integration/fuse/bin/alluxio-fuse",
 		fmt.Sprintf("lib/alluxio-underfs-abfs-%v.jar", version),
 		fmt.Sprintf("lib/alluxio-underfs-adl-%v.jar", version),
+		fmt.Sprintf("lib/alluxio-underfs-hcfs-cephfs-%v.jar", version),
```
Comment (apc999) [inline dev/scripts/src/alluxio.org/build-distribution/cmd/generate-tarball.go:219]: sort alphabetically

File: shaded/hcfs-cephfs/pom.xml [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+  <packaging>jar</packaging>
+  <name>Alluxio Shaded Libraries - Hcfscephfs</name>
+  <description>Shaded hcfs-cephfs Module</description>
+
+  <properties>
+    <!-- The build path need to be defined here as well as in the parent pom so that mvn can -->
+    <!-- run properly from sub-project directories -->
+    <build.path>${project.parent.parent.basedir}/build</build.path>
+    <ufs.hcfscephfs.version>0.80.6</ufs.hcfscephfs.version>
+    <shading.prefix>alluxio.shaded.hdfs</shading.prefix>
```
Comment (apc999) [inline shaded/hcfs-cephfs/pom.xml:30]: `alluxio.shaded.hdfs` ?

File: underfs/hcfs-cephfs/src/main/java/alluxio/underfs/hcfscephfs/HcfsCephfsUnderFileSystemFactory.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+import alluxio.underfs.hdfs.HdfsUnderFileSystemFactory;
+import alluxio.underfs.UnderFileSystemConfiguration;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Factory for creating {@link HdfsUnderFileSystem}.
+ *
+ * It caches created {@link HdfsUnderFileSystem}s, using the scheme and authority pair as the key.
+ */
```
Comment (apc999) [inline underfs/hcfs-cephfs/src/main/java/alluxio/underfs/hcfscephfs/HcfsCephfsUnderFileSystemFactory.java:25]: update javadoc

File: shaded/hcfs-cephfs/pom.xml (hunk 4 above)
Comment (maobaolong) [reply shaded/hcfs-cephfs/pom.xml:30]: Yeah, we shade it for keeping the package name consistent with shaded/hdfs

File: underfs/hcfs-cephfs/src/main/java/alluxio/underfs/hcfscephfs/HcfsCephfsUnderFileSystemFactory.java (hunk 5 above)
Comment (maobaolong) [reply underfs/hcfs-cephfs/src/main/java/alluxio/underfs/hcfscephfs/HcfsCephfsUnderFileSystemFactory.java]: I think this document can make sense, actually, we have no CephfsUnderFileSystem, so the HcfsCephfsUnderFileSystemFactory still create HdfsUnderFileSystem to talk to cephfs.

File: dev/scripts/src/alluxio.org/build-distribution/cmd/generate-tarball.go (hunk 3 above)
Comment (maobaolong) [reply dev/scripts/src/alluxio.org/build-distribution/cmd/generate-tarball.go]: done

File: underfs/pom.xml (hunk 2 above)
Comment (maobaolong) [reply underfs/pom.xml]: done

File: shaded/pom.xml (hunk 1 above)
Comment (maobaolong) [reply shaded/pom.xml]: done

**Comment by maobaolong** [comment]
@apc999 Thanks for your review, please take another look.

**Review by apc999 — APPROVED** [review]

**Comment by apc999** [comment]
alluxio-bot, merge this please

**Pull request merged**